modules/azure/storage.go

```diff
@@ -0,0 +1,246 @@
+// This function would fail the test if there is an error.
+func GetStorageAccountPropertyE(storageAccountName, resourceGroupName, subscriptionID string) (*storage.Account, error) {
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	resourceGroupName, err2 := getTargetAzureResourceGroupName((resourceGroupName))
+	if err2 != nil {
+		return nil, err2
+	}
+	client, err := GetStorageAccountClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	account, err := client.GetProperties(context.Background(), resourceGroupName, storageAccountName, "")
+	if err != nil {
+		return nil, err
+	}
+	return &account, nil
+}
+
+// GetStorageAccountClientE creates a storage account client.
+// This function would fail the test if there is an error.
+func GetStorageAccountClientE(subscriptionID string) (*storage.AccountsClient, error) {
+	storageAccountClient := storage.NewAccountsClient(os.Getenv(AzureSubscriptionID))
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+	storageAccountClient.Authorizer = *authorizer
+	return &storageAccountClient, nil
+}
+
+// GetStorageBlobContainerClientE creates a storage container client.
+// This function would fail the test if there is an error.
+func GetStorageBlobContainerClientE(subscriptionID string) (*storage.BlobContainersClient, error) {
+	blobContainerClient := storage.NewBlobContainersClient(os.Getenv(AzureSubscriptionID))
+	authorizer, err := NewAuthorizer()
+
+	if err != nil {
+		return nil, err
+	}
+	blobContainerClient.Authorizer = *authorizer
+	return &blobContainerClient, nil
+}
+
+// GetStorageURISuffixE returns the proper storage URI suffix for the configured Azure environment.
+// This function would fail the test if there is an error.
+func GetStorageURISuffixE() (*string, error) {
+	var envName = "AzurePublicCloud"
+	env, err := azure.EnvironmentFromName(envName)
+	if err != nil {
+		return nil, err
+	}
+	return &env.StorageEndpointSuffix, nil
+}
+
+// GetStorageAccountPrimaryBlobEndpointE gets the storage account blob endpoint as URI string.
+// This function would fail the test if there is an error.
+func GetStorageAccountPrimaryBlobEndpointE(storageAccountName, resourceGroupName, subscriptionID string) (string, error) {
+	storageAccount, err := GetStorageAccountPropertyE(storageAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+
+	return *storageAccount.AccountProperties.PrimaryEndpoints.Blob, nil
+}
+
+// GetStorageDNSStringE builds and returns the storage account dns string if the storage account exists.
+// This function would fail the test if there is an error.
+func GetStorageDNSStringE(storageAccountName, resourceGroupName, subscriptionID string) (string, error) {
+	retval, err := StorageAccountExistsE(storageAccountName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", err
+	}
+	if retval {
+		storageSuffix, _ := GetStorageURISuffixE()
```

Comment:
Gruntworks guidance is to surface errors from E methods.  We should check if err != nil before returning formatted value.